src/init.cpp

```diff
@@ -1713,7 +1713,23 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     connOptions.m_msgproc = node.peerman.get();
     connOptions.nSendBufferMaxSize = 1000 * args.GetIntArg("-maxsendbuffer", DEFAULT_MAXSENDBUFFER);
     connOptions.nReceiveFloodSize = 1000 * args.GetIntArg("-maxreceivebuffer", DEFAULT_MAXRECEIVEBUFFER);
-    connOptions.m_added_nodes = args.GetArgs("-addnode");
+
+    std::vector<AddedNodeParams> added_nodes;
+    for (auto added_node : args.GetArgs("-addnode")) {
+        ConnectionType conn_type = ConnectionType::MANUAL;
+        const size_t index = added_node.rfind('=');
+        if (index != std::string::npos) {
+            const std::string conn_name = added_node.substr(index + 1);
+            if (conn_name == ConnectionTypeAsString(ConnectionType::MANUAL_BLOCK_RELAY)) {
+                added_node = added_node.substr(0, index);
+                conn_type = ConnectionType::MANUAL_BLOCK_RELAY;
+            } else {
```

Comment:
Maybe also add the case for `ConnectionType::MANUAL`?

`./src/bitcoind -addnode='test.local=manual'`
> 2023-01-16T16:04:19Z Error: Invalid -addnode connection type: manual
> Error: Invalid -addnode connection type: manual

Reply:
Not sure about this: Currently, the help for "addnode" states that only `=manual-block-relay` is possible (manual is the default). Adding this to the help might only complicate things for the user, because there would be no difference between `=manual` and no argument at all?

Reply:
I prefer not having `=manual`, it would add more stuff to the code and I believe it might complicate for the user, it's the default behavior, not sure why someone would want to specify it.